linkerd/protocol/h2/src/main/scala/com/twitter/finagle/buoyant/h2/LinkerdHeaders.scala

```diff
@@ -74,6 +74,20 @@ object LinkerdHeaders {
           deadline.andThen(dtab).andThen(next)
       }
 
+    val clearServerModule: Stackable[ServiceFactory[Request, Response]] =
+      new Stack.Module0[ServiceFactory[Request, Response]] {
+        val role = serverModule.role
+        val description = "Clears linkerd context from http request headers"
+
+        val deadline = new Deadline.ClearServerFilter
+        val dtab = new Dtab.ClearServerFilter
+        val sample = new Trace.ClearServerFilter
+        val trace = new Trace.ClearServerFilter
```

Comment:
Yeah, I as thinking about that... As much as I was trying to avoid iterating through headers, I think it's probably preferable to do it (additionally, after the targeted context clears have been performed)